documentation/modules/cruise-control/con-optimization-proposals.adoc

```diff
@@ -0,0 +1,91 @@
+// Module included in the following assemblies:
+//
+// assembly-cruise-control-concepts.adoc
+
+[id='con-optimization-proposals-{context}']
+
+= Optimization proposals overview
+
+An _optimization proposal_ is a summary of proposed changes that would produce a more balanced Kafka cluster, with partition workloads distributed more evenly among the brokers. 
+Each optimization proposal is based on the set of xref:con-optimization-goals-{context}[optimization goals] that were used to generate it.
+
+Use the summary (shown in the `Status` of the `KafkaRebalance` resource) to decide whether to implement the cluster rebalance, or change the optimization goals.
+
+NOTE: All optimization proposals are dry runs: you cannot implement a cluster rebalance without first generating an optimization proposal. 
+There is no limit to the number of optimization proposals that can be generated.
+
+[discrete]
+== Cached optimization proposal
+
+Cruise Control maintains a _cached optimization proposal_ based on the configured default optimization goals.
+Generated from the workload model, the cached optimization proposal is updated every 15 minutes to reflect the current state of the Kafka cluster.
+If you generate an optimization proposal using the default optimization goals, Cruise Control returns the most recent cached proposal.
+
+To change the cached optimization proposal refresh interval, edit the `proposal.expiration.ms` setting in the Cruise Control deployment configuration.
+Consider a shorter interval for fast changing clusters, although this increases the load on the Cruise Control server.
+
+[id="contents-optimization-proposals"]
+[discrete]
+== Contents of optimization proposals
+
+The following table explains the properties contained in an optimization proposal:
+
+[cols="2*",options="header",stripes="none",separator=¦]
+|===
+
+m¦JSON property
+¦Description
+
+m¦numIntraBrokerReplicaMovements
+¦The total number of partition replicas that will be transferred between the disks of the cluster's brokers.
+
+*Performance impact during rebalance operation*: Relatively high, but lower than `numReplicaMovements`.
+
+m¦excludedBrokersForLeadership
+¦Not yet supported. An empty list is returned.
+
+m¦numReplicaMovements
+¦The number of partition replicas that will be moved between separate brokers.
```

Comment:
@tomncooper are we confident that this is interbroker movement, and not the sum of interbroker and intrabroker?

Reply:
I will double check the CC source code, just to make sure.

Reply:
That number comes from [this method](https://github.com/linkedin/cruise-control/blob/e5fdfe8df81957e69fd50b9e290e72e45d26c812/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/OptimizerResult.java#L232) and is a count incremented in an `if` branch for removal or addition of replicas to a broker which is separate from the `if` branch which increments the **intra**broker (between disks) count, so I pretty confident this is the **inter**broker replica movement count.

Reply:
OK, I'll leave this unchanged.